### ISSUE TITLE
Data trees in RDF 

### ISSUE THREAD

**Issue by DiellzaElshani** [BHoM/RDF_Prototypes#46]
#### Description:

BHoM allows data tree as input in custom objects, however we did not provide a translation of data trees in RDF. Currently they are translated to Base64JsonSerialized. 
Is there any anyway to represent data trees in RDF @danielhz ?

**Comment by DiellzaElshani** [comment]
ps: to add, a data tree is a graph basically, we just need to explode the information within it.

**Comment by danielhz** [comment]
Have these threes labelled edges?

**Comment by alelom** [comment]
> Have these threes labelled edges? 

Yes, with an identifier in the form of `{branch; sub-branch; sub-sub-branch; ... }`:
![image](https://user-images.githubusercontent.com/6352844/182590550-5489b416-011d-4f61-bb06-72b41fdd9ce2.png)
![image](https://user-images.githubusercontent.com/6352844/182590883-61a3981f-16b3-408d-a1ba-3c985abea7eb.png)



> Does this tree have a defined root? 

Not sure what do you mean by defined root, but yes, the label `{0}` will always have the source corresponding to the root.

> Are siblings ordered?

Yes, see images above.

**Comment by danielhz** [comment]
These trees can be represented with collections or with containers. The following are two ways to represent the two trees in the first figure.

### With `rdf:Sec`

```ttl
:tree1 a bhom:Tree ;
  bhom:treeValues [ a rdf:Seq ; rdf:_1 :u1 ; rdf:_2 :u2; rdf:_3 :u3 ; rdf:_4 :u4 ] .

:tree a bhom:Tree ;
  bhom:treeValues [ rdf:Seq ;
    rdf:_1 [ a rdf:Seq ; rdf:_1 :u1 ; rdf:_2 :u2 ] ;
    rdf:_2 [ a rdf:Seq ; rdf:_1 :u3 ; rdf:_2 :u4 ] ] .
```

### With `rdf:Collection`

```ttl
:tree1 a bhom:Tree ;
  bhom:treeValues ( :u1 :u2 :u3 :u4 ) .

:tree a bhom:Tree ;
  bhom:threeValues ( ( :u1 :u2 )  ( :u3 :u4 ) ) .
```
